matchzoo/preprocessors/basic_preprocessor.py

```diff
@@ -0,0 +1,121 @@
+"""Basic Preprocessor."""
+
+import logging
+
+from tqdm import tqdm
+
+from matchzoo import engine, processor_units
+from matchzoo import DataPack
+from matchzoo import chain_transform, build_vocab_unit, \
+    build_unit_from_data_pack
+
+logger = logging.getLogger(__name__)
+tqdm.pandas()
+
+
+class BasicPreprocessor(engine.BasePreprocessor):
+    """
+    Baisc preprocessor helper.
+
+    :param fixed_length_left: Integer, maximize length of :attr:`left` in the
+        data_pack.
+    :param fixed_length_right: Integer, maximize length of :attr:`right` in the
+        data_pack.
+
+    Example:
+        >>> import matchzoo as mz
+        >>> train_data = mz.datasets.toy.load_data('train')
+        >>> test_data = mz.datasets.toy.load_data('test')
+        >>> preprocessor = mz.preprocessors.BasicPreprocessor(
+        ...     fixed_length_left=10,
+        ...     fixed_length_right=20
+        ... )
+        >>> preprocessor = preprocessor.fit(train_data)
+        >>> preprocessor.context['input_shapes']
+        [(10,), (20,)]
+        >>> preprocessor.context['vocab_size']
+        284
+        >>> processed_train_data = preprocessor.transform(train_data)
+        >>> type(processed_train_data)
+        <class 'matchzoo.data_pack.data_pack.DataPack'>
+        >>> test_data_transformed = preprocessor.transform(test_data)
+        >>> type(test_data_transformed)
+        <class 'matchzoo.data_pack.data_pack.DataPack'>
+
+    """
+
+    def __init__(self, fixed_length_left: int = 30,
+                 fixed_length_right: int = 30):
+        """Initialization."""
+        super().__init__()
+        self._fixed_length_left = fixed_length_left
+        self._fixed_length_right = fixed_length_right
+        self._left_fixedlength_unit = processor_units.FixedLengthUnit(
+            self._fixed_length_left, pad_mode='post')
+        self._right_fixedlength_unit = processor_units.FixedLengthUnit(
+            self._fixed_length_right, pad_mode='post')
+
+    def fit(self, data_pack: DataPack, verbose=1):
+        """
+        Fit pre-processing context for transformation.
+
+        :param data_pack: data_pack to be preprocessed.
+        :param verbose: Verbosity.
+        :return: class:`DRMMTKSPreprocessor` instance.
```

Comment:
Why is the return type `DRMMTKSPreprocessor` insteads of the type for the basic preprocessor ?